osxkeychain/osxkeychain.go

```diff
@@ -46,6 +46,13 @@ func (h Osxkeychain) Add(creds *credentials.Credentials) error {
 	item.SetLabel(credentials.CredsLabel)
 	item.SetAccount(creds.Username)
 	item.SetData([]byte(creds.Secret))
+	// Prior to v0.9.0, the credential helper was searching for credentials with
+	// the "dflt" authentication type. Since this version, Get doesn't use that
+	// attribute anymore.
+	//
+	// In order to keep compatibility with older versions, we need to store
+	// credentials with this attribute set.
+	item.SetAuthenticationType("dflt")
```

Comment:
As a follow-up, it'd be great to add this constant and the ones I hardcoded in https://github.com/docker/docker-credential-helpers/pull/361 to `keybase/go-chain`.

I started doing that, but it's lower priority than getting this fixed here before we release the next version of Docker Desktop with a creds store that doesn't support downgrades.

Reply:
Hum yes indeed I looked at previous implem https://github.com/docker/docker-credential-helpers/pull/282/files#diff-13e5d0d1bee45d43439783edc43138e1d9584d2ec29cab4969460384db72ec42L28 and we set it to `dflt` as well.

I thought `dflt` was the default SecAuthenticationType in `keybase/go-keychain` so agree to set this upstream.

Reply:
Oh! I forgot to add a link to `keychain_get` in my comment. Let me update it before I merge.